### PR TITLE
[WIP] Issue 158

### PR DESCRIPTION
**Pull request by Frankkwizera**
Adding options of creating new sites and editing existing ones.

-> Still working on it .

### CODE REVIEW

File: seshdash/views.py [hunk 1]
```diff
@@ -782,3 +781,43 @@ def graphs(request):
         return HttpResponseBadRequest()
 
 
+#function to create/edit sites
+@login_required
+def edit_settings(request):
+    site_available = []
+    #fetching list of all site
```
Comment (jlark) [inline seshdash/views.py]: carefull. Not all sites. Only sites for user!

File: seshdash/views.py [hunk 2]
```diff
@@ -782,3 +781,43 @@ def graphs(request):
...
+
+    if request.method == 'POST':
+
+        form = SiteForm(request.POST)
+        #checking if the form is valid
+        if form.is_valid:
+
+            form = form.save()
+        else:
+            return HttpResponse("Didn`t validate")
```
Comment (jlark) [inline seshdash/views.py]: For the response you should return the form again. The form will automatically modified to show what the errors are in it.

File: seshdash/static/sesh-ui/js/sesh_dashboard.js [hunk 3]
```diff
@@ -249,6 +249,41 @@ Morris.Bar({
...
 
-  $("#id_location_city").keypress(function(){
-                    $("#id_location_country").val($(this).val());
-                  });
+
+/*------------------------------------------------------------------------*/
+$(document).ready(function(){
+            //$("#create_site").tabs();
+
+            $('.edit-site').hide();
```
Comment (jlark) [inline seshdash/static/sesh-ui/js/sesh_dashboard.js]: Is there a reason you're doing all the navigation in javascript?

File: seshdash/static/sesh-ui/js/graphs.js [hunk 4]
```diff
@@ -67,10 +75,12 @@ function daily_data_points_graph() {
    time = "24h";
   message = ('Daily ' + drop_choice1 + '  With  ' + drop_choice2);
   $("#title-message").html(message);
+   $("#dynamic_graph").hide();
+
   $.post("/graphs",{csrfmiddlewaretoken: csrftoken, 'choice': [drop_choice1,drop_choice2], 'time':time , 'active_site_id':active_site_id},function(data){
-        
+
+        var response = JSON.parse(data);
```
Comment (jlark) [inline seshdash/static/sesh-ui/js/graphs.js]: Should this be happening twice?

File: seshdash/views.py [hunk 5]
```diff
@@ -782,3 +781,83 @@ def graphs(request):
         return HttpResponseBadRequest()
 
 
+#function to editing existing sites
+@login_required
+def edit_settings(request):
+    all_sites = []
+    #all sesh sites
+    sites = Sesh_Site.objects.all()
```
Comment (jlark) [inline seshdash/views.py]: get only user sites.

File: seshdash/views.py [hunk 6]
```diff
@@ -782,3 +781,83 @@ def graphs(request):
...
+
+    for site in sites:
+
+        all_sites.append({site.site_name:site.id})
+
+    SiteName = request.GET
```
Comment (jlark) [inline seshdash/views.py]: make sure to check there is actually a GET request

File: seshdash/views.py [hunk 7]
```diff
@@ -782,3 +781,83 @@ def graphs(request):
         return HttpResponseBadRequest()
 
 
+#function to editing existing sites
+@login_required
+def edit_settings(request):
```
Comment (jlark) [inline seshdash/views.py]: let's rename this function `edit_site` since the other one is `add_site`

File: seshdash/views.py [hunk 8]
```diff
@@ -782,3 +781,83 @@ def graphs(request):
...
+    all_sites = []
```
Comment (jlark) [inline seshdash/views.py]: do we really need another array? Why not just use the `sites` array?

File: seshdash/static/sesh-ui/js/graphs.js (hunk 4 above)
Comment (Frankkwizera) [inline seshdash/static/sesh-ui/js/graphs.js]: mistake

File: local [hunk 9]
```diff
@@ -0,0 +1 @@
+Subproject commit 5b7401d4cb0d3801f9ef4c011ae0642dd65483a0
```
Comment (jlark) [inline local]: I think this file is an accident?

File: seshdash/tests/test_settings_buttons.py [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+                              'position_1':42,
+                              'installed_kw':2,
+                              'system_voltage':4,
+                              'number_of_panels':100,
+                              'battery_bank_capacity':1200}
+        form = SiteForm(data)
+        self.assertTrue(form.is_valid())
+        form.save()
+        response = f.post('/add_site', data)
+        self.assertEqual(response.status_code, 200)
```
Comment (jlark) [inline seshdash/tests/test_settings_buttons.py]: This test is incomplete are you checking that the site was actually created in the database?

File: seshdash/tests/test_settings_buttons.py [hunk 11]
```diff
@@ -0,0 +1,101 @@
...
+                              'location_country':u'rwanda',
+                              'time_zone':'Africa/Kigali',
+                              'position_0':36,
+                              'position_1':42,
+                              'installed_kw':2,
+                              'system_voltage':4,
+                              'number_of_panels':100,
+                              'battery_bank_capacity':1200}
+        form = SiteForm(data)
+        self.assertTrue(form.is_valid())
```
Comment (jlark) [inline seshdash/tests/test_settings_buttons.py]: Where is the test for editing the site?

File: seshdash/views.py [hunk 12]
```diff
@@ -713,10 +712,11 @@ def get_latest_bom_data(request):
 def search(request):
     data=[]
     sites = Sesh_Site.objects.all()
```
Comment (jlark) [inline seshdash/views.py]: This is a bug. You should only be searching sites the user has permission to access.

File: seshdash/views.py [hunk 13]
```diff
@@ -806,3 +806,44 @@ def graphs(request):
...
+    user_site_name = []
+    user_site_id = []
+    sites =  _get_user_sites(request)
+    for site in sites:
+        user_site_name.append(site.site_name)
+        user_site_id.append(site.id)
+    user_sites = dict(zip(user_site_id,user_site_name))
+    # on ajax 
+    if request.method == 'POST':
+
```
Comment (jlark) [inline seshdash/views.py]: Please get rid of unnecessary white spaces. Use whitespaces to break up the code into logical sections.

File: seshdash/views.py [hunk 14]
```diff
@@ -806,3 +806,44 @@ def graphs(request):
         return HttpResponse(json.dumps(results))
     else:
         return HttpResponseBadRequest()
+
+#function to editing existing sites
+@login_required
+def edit_site(request,site_Id):
+    #creating an instance to populate a form
+    instance = get_object_or_404(Sesh_Site, id=site_Id)
```
Comment (jlark) [inline seshdash/views.py]: what if site_id is sent empty? What happens?

File: seshdash/tests/test_settings_buttons.py (hunk 11 above)
Comment (Frankkwizera) [inline seshdash/tests/test_settings_buttons.py]: oops forgot it

File: local (hunk 9 above)
Comment (Frankkwizera) [inline local]: how?

File: seshdash/tests/test_settings_buttons.py (hunk 10 above)
Comment (Frankkwizera) [inline seshdash/tests/test_settings_buttons.py]: done

File: seshdash/views.py (hunk 12 above)
Comment (Frankkwizera) [inline seshdash/views.py]: done

File: seshdash/views.py (hunk 13 above)
Comment (Frankkwizera) [inline seshdash/views.py]: done!

File: seshdash/views.py (hunk 14 above)
Comment (Frankkwizera) [inline seshdash/views.py]: done

**Comment by jlark** [comment]
Thanks @Frankkwizera is can you commit the updates so I can review?

**Comment by jlark** [comment]
@Frankkwizera make sure your logs folder is not in the commit. if it's getting commmited to a pull request and make sure you have the latest .gitignore.

Also please remove the local folder

**Comment by jlark** [comment]
Builds are failing, please correct @Frankkwizera

**Comment by jlark** [comment]
Make sure the /logs folder is there but that all the files in it get ignored.

**Pull request merged**